crates/cli-support/src/js/mod.rs

```diff
@@ -1769,15 +1767,21 @@ impl<'a> Context<'a> {
         self.memview("Float64", memory)
     }
 
+    fn expose_dataview_memory(&mut self, memory: MemoryId) -> MemView {
+        self.memview("DataView", memory)
+    }
+
     fn memview(&mut self, kind: &'static str, memory: walrus::MemoryId) -> MemView {
         let view = self.memview_memory(kind, memory);
         if !self.should_write_global(view.name.clone()) {
             return view;
         }
         let mem = self.export_name_of(memory);
 
+        let is_dataview = kind == "DataView";
+
         let cache = format!("cached{}Memory{}", kind, view.num);
-        let resized_check = if self.module.memories.get(memory).shared {
+        let resized_check = if self.module.memories.get(memory).shared || is_dataview {
```

Comment:
This should be explained in a comment.
Additionally, I think we can do a cheaper check by testing [`ArrayBuffer.prototype.detached`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer/detached).

Reply:
Yes, we check its detached status, but unfortunately, node.js currently does not support this.

Reply:
Yes, apparently this is reproduced in CI as well.
Still trying to figure out the details, because I'm unable to re-produce it locally.

Reply:
<img width="1133" alt="image" src="https://github.com/user-attachments/assets/64c2a0d0-6462-4381-ac2e-2e5049eb4abb">

Reply:
Yes, so the issue is that Node v20 is broken in this regard, I will skim their changelog.
But in any case, the current LTS version is v20, so we would want to support that.

I found that Node v22 works.

Reply:
This doesn't just affect Node, Safari also only implemented this since v17.4 and we definitely support older versions then that.

So its good that CI caught it, I added a bit of an ugly fallback, but it works.